### PR TITLE
Module updates/addition of a vendor directory

### PR DESCRIPTION
**Pull request by cfergeau**
This PR takes advantage of the recent machine changes to remove most machine-driver-libvirt
dependencies, and adds a vendor/ directory.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 *.sw*
 crc-driver-libvirt-*
-vendor/
```
Comment (praveenkumar) [inline .gitignore:3]: Can we also have a  make target to sync the vendor and have a check similar to what we have in crc to make sure it never become out of sync?

**Comment by guillaumerose** [comment]
Can we have this https://github.com/code-ready/machine/pull/54 merged before this one?

**Comment by cfergeau** [comment]
> Can we have this [code-ready/machine#54](https://github.com/code-ready/machine/pull/54) merged before this one?

Ah sure, it's merged now, so I'll update the branch in a bit.

File: .gitignore (hunk 1 above)
Comment (cfergeau) [reply .gitignore:3]: I added `make vendorcheck`, I'm unconvinced about adding `make vendor`, as it's just calling `go mod vendor`

Comment (cfergeau) [reply .gitignore:3]: > > Can we have this [code-ready/machine#54](https://github.com/code-ready/machine/pull/54) merged before this one?
> 
> Ah sure, it's merged now, so I'll update the branch in a bit.

Should be ok now.

**Comment by guillaumerose** [comment]
CI is unhappy, `go mod vendor` perhaps?

**Comment by cfergeau** [comment]
> CI is unhappy, `go mod vendor` perhaps?

the `verify-vendor.sh` script needs more tools to be installed, adding these to the Containerfiles
edit: only `diffutils` was missing, green now.

**Review by guillaumerose — APPROVED** [review]

**Pull request merged**